virtcontainers/clh.go

```diff
@@ -165,98 +136,77 @@ func (clh *cloudHypervisor) createSandbox(ctx context.Context, id string, networ
 	}
 
 	// Set initial memomory size of the cloud hypervisor
-	clh.cliBuilder.SetMemory(&CLIMemory{
-		memorySize:  clh.config.MemorySize,
-		backingFile: "/dev/shm",
-	})
+	clh.vmconfig.Memory.Size = int64(clh.config.MemorySize) << utils.MibToBytesShift
+	clh.vmconfig.Memory.File = "/dev/shm"
 	// Set initial amount of cpu's for the cloud hypervisor
-	clh.cliBuilder.SetCpus(&CLICpus{
-		cpus: clh.config.NumVCPUs,
-	})
+	clh.vmconfig.Cpus = chclient.CpuConfig{
+		// cast to int32, this is a openAPI limitation that not support unsigned values
```

Comment:
Does a negative number of CPUs make sense here, anyway? If we want to still add this, suggestion:

```suggestion
		// cast to int32, as openAPI has a limitation that it does not support unsigned values 
```

Reply:
It does not make sense but a limitation with the API today